x-pack/plugins/enterprise_search/public/applications/app_search/components/settings/log_retention/log_retention_logic.ts

```diff
@@ -0,0 +1,128 @@
+import { flashAPIErrors } from '../../../../shared/flash_messages';
+import { convertLogRetentionFromServerToClient } from './utils/convert_log_retention';
+
+interface ILogRetentionActions {
+  clearLogRetentionUpdating(): { value: boolean };
+  closeModals(): { value: boolean };
+  fetchLogRetention(): { value: boolean };
+  saveLogRetention(
+    option: ELogRetentionOptions,
+    enabled: boolean
+  ): { option: ELogRetentionOptions; enabled: boolean };
+  setOpenModal(option: ELogRetentionOptions): { option: ELogRetentionOptions };
+  toggleLogRetention(option: ELogRetentionOptions): { option: ELogRetentionOptions };
+  updateLogRetention(logRetention: ILogRetention): { logRetention: ILogRetention };
+}
+
+interface ILogRetentionValues {
+  logRetention: ILogRetention | null;
+  logsRetentionUpdating: boolean;
+  openModal: ELogRetentionOptions | null;
+}
+
+export const LogRetentionLogic = kea<MakeLogicType<ILogRetentionValues, ILogRetentionActions>>({
+  path: ['enterprise_search', 'app_search', 'log_retention_logic'],
+  actions: () => ({
+    clearLogRetentionUpdating: true,
+    closeModals: true,
+    fetchLogRetention: true,
+    saveLogRetention: (option, enabled) => ({ enabled, option }),
+    setOpenModal: (option) => ({ option }),
+    toggleLogRetention: (option) => ({ option }),
+    updateLogRetention: (logRetention) => ({ logRetention }),
+  }),
+  reducers: () => ({
+    logRetention: [
+      null,
+      {
+        updateLogRetention: (previousValue, { logRetention }) => {
+          return {
+            [ELogRetentionOptions.Analytics]: {
+              ...previousValue?.[ELogRetentionOptions.Analytics],
+              ...logRetention[ELogRetentionOptions.Analytics],
+            },
+            [ELogRetentionOptions.API]: {
+              ...previousValue?.[ELogRetentionOptions.API],
+              ...logRetention[ELogRetentionOptions.API],
+            },
+          };
+        },
+      },
+    ],
+    logsRetentionUpdating: [
+      false,
+      {
+        clearLogRetentionUpdating: () => false,
+        closeModals: () => false,
+        fetchLogRetention: () => true,
+        toggleLogRetention: () => true,
+      },
+    ],
+    openModal: [
+      null,
+      {
+        closeModals: () => null,
+        saveLogRetention: () => null,
+        setOpenModal: (_, { option }) => option,
+      },
+    ],
+  }),
+  listeners: ({ actions, values }) => ({
+    fetchLogRetention: async () => {
+      try {
+        const { http } = HttpLogic.values;
+        const response = await http.get('/api/app_search/log_settings');
+        actions.updateLogRetention(
+          convertLogRetentionFromServerToClient(response as ILogRetentionServer)
+        );
+      } catch (e) {
+        flashAPIErrors(e);
+      } finally {
+        actions.clearLogRetentionUpdating();
+      }
+    },
+    saveLogRetention: async ({ enabled, option }) => {
+      const updateData = { [option.toString()]: { enabled } };
```

Comment:
Is there a realistic scenario where `option` is not already a string? just wondering/curious

Reply:
It's an enum, so I think you're right, it's unnecessary to call `.toString`

Reply:
https://github.com/elastic/kibana/pull/82364/commits/6b0989e811c6d65ac412997c5fdd9c720e0a15be